Options.py

```diff
@@ -1115,6 +1115,97 @@ def verify(self, world: typing.Type[World], player_name: str, plando_options: "P
+    locations: typing.Union[typing.List[str]]
+    world: typing.Union[int, str, bool, None, typing.Iterable[str], typing.Set[int]] = False
+    from_pool: bool = True
+    force: typing.Union[bool, typing.Literal["silent"]] = "silent"
+    count: typing.Union[int, bool, typing.Dict[str, int]] = False
+    percentage: int = 100
+
+
+class PlandoItems(Option[typing.List[PlandoItem]]):
+    """Generic items plando."""
+    default = ()
+    supports_weighting = False
+    display_name = "Plando Items"
+
+    def __init__(self, value: typing.Iterable[PlandoItem]) -> None:
+        self.value = list(deepcopy(value))
+        super().__init__()
+
+    @classmethod
+    def from_any(cls, data: typing.Any) -> Option[typing.List[PlandoItem]]:
+        if not isinstance(data, typing.Iterable):
+            raise Exception(f"Cannot create plando items from non-Iterable type, got {type(data)}")
+
+        value: typing.List[PlandoItem] = []
+        for item in data:
+            if isinstance(item, typing.Mapping):
+                percentage = item.get("percentage", 100)
+                if random.random() < float(percentage / 100):
+                    count = item.get("count", False)
+                    items = item.get("items", [])
+                    if not items:
+                        items = item.get("item", None)  # explcitly throw an error here if not present
+                        if not items:
+                            raise Exception("You must specify at least one item to place items with plando.")
+                        items = [items]
+                    locations = item.get("locations", [])
+                    if not locations:
+                        locations = item.get("location", None)
+                        if not locations:
+                            raise Exception("You must specify at least one location to place items with plando.")
+                        locations = [locations]
+                    world = item.get("world", False)
+                    from_pool = item.get("from_pool", True)
+                    force = item.get("force", "silent")
+                    value.append(PlandoItem(items, locations, world, from_pool, force, count, percentage))
+            elif isinstance(item, PlandoItem):
+                if random.random() < float(item.percentage / 100):
+                    value.append(item)
+            else:
+                raise Exception(f"Cannot create plando item from non-Dict type, got {type(item)}.")
+        return cls(value)
+
+    def verify(self, world: typing.Type[World], player_name: str, plando_options: "PlandoOptions") -> None:
+        from BaseClasses import PlandoOptions
+        if not (PlandoOptions.items & plando_options):
+            # plando is disabled but plando options were given so overwrite the options
+            self.value = []
+            logging.warning(f"The plando items module is turned off, "
```

Comment:
Two things:
1. This warning always triggers, even if you just have `plando_items: []`
2. This warning looks strange with player names using the `{}` methods. For example, this can be printed:
> The plando items module is turned off, so items for Player{number} will be ignored.

Reply:
So interestingly, this only occurs for when `plando_items: []` is present. If you remove the plando items lines entirely, it skips verify. Fixed.

As for the warning, this code runs before `{}` methods in names are evaluated, so there's nothing that can really be done there.